src/routing_table/network_tests.rs

```diff
@@ -318,16 +321,17 @@ pub fn verify_network_invariant<'a, T, U>(nodes: U)
                 node.sections[&prefix].clone()
             };
             if let Some(&mut (ref mut src, ref mut section)) = sections.get_mut(&prefix) {
-                assert!(*section == section_content,
-                        "Section with prefix {:?} doesn't agree between nodes {:?} and {:?}\n\
-                        {:?}: {:?}, {:?}: {:?}",
-                        prefix,
-                        node.our_name,
-                        src,
-                        node.our_name,
-                        section_content,
-                        src,
-                        section);
+                assert_eq!(*section,
+                           section_content,
+                           "Section with prefix {:?} doesn't agree between nodes {:?} and {:?}\n\
+                            {:?}: {:?}, {:?}: {:?}",
+                           prefix,
+                           node.our_name,
+                           src,
+                           node.our_name,
+                           section_content,
+                           src,
+                           section);
```

Comment:
Yes, they would.  To be honest, I hardly ever use the actual contents of the sections shown in that message - I just run the test with logging and have a look at the two RTs indicated by `node.name` and `src`.  I could remove the last four variables from the formatted string if that suits everyone else?

If not, I think we're possibly better keeping the log message as is, (since it attributes the two sections to the relevant owners).  In that case, I can leave the PR as it is now, or revert back to `assert!()` and allow the clippy lint for that function?